src/monitors.ts

```diff
@@ -244,16 +246,23 @@ export function replaceCriticalThreshold(query: string, criticalThreshold: numbe
 }
 
 /**
- * Helper function that interpolates template variables, e.g. replaces template
- * variable `$functionName` with its default value `{{functionname.name}}`.
- * @param templateVariables - Template variables defined in the monitor JSON
+ * Helper function that interpolates template variables (if any), e.g.
+ * replaces template variable `$functionName` with its default value
+ * `{{functionname.name}}`. Template variables with no default value will be
+ * ignored.
+ * @param recommendedMonitors  - recommended monitors
  * @param name - The string that may contain uninterpolated template variables,
  *        e.g. "High Error Rate on $functionName in $regionName for $awsAccount"
  * @returns Interpolation result, e.g. "High Error Rate on {{functionname.name}} in {{region.name}} for {{aws_account.name}}"
  */
-function interpolateTemplateVariables(templateVariables: TemplateVariable[], name: string): string {
-  templateVariables.forEach((templateVariable) => {
-    name = name.replace("$" + templateVariable.name, templateVariable.defaults[0]);
-  });
+function interpolateTemplateVariables(recommendedMonitor: ServerlessMonitor, name: string): string {
```

Comment:
non-blocking nit: should we call the parameter something like `baseString` instead of `name` in case we later want to interpolate other parts of the monitor, such as the `message` field?

Reply:
I had thought about a better name for "name" but couldn't find one. Thanks for the suggestion!